File: src/templates/nestjs/scaffold/src/backend/src/health/health.controller.ts
import { Controller, Get } from "@nestjs/common";

import { HealthService } from "./health.service.js";

// Thin: delegates to the provider and returns its value (Nest serializes).
@Controller("health")
export class HealthController {
  constructor(private readonly health: HealthService) {}

  @Get()
  check() {
    return this.health.status();
  }
}

File: src/templates/spring-boot/scaffold/src/backend/src/main/java/com/example/app/health/HealthController.java
package com.example.app.health;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

// Thin: maps the route and delegates to the service (Spring serializes the value).
@RestController
@RequestMapping("/health")
public class HealthController {

  private final HealthService health;

  public HealthController(HealthService health) {
    this.health = health;
  }

  @GetMapping
  public HealthStatus check() {
    return health.status();
  }
}
